koboldcpp having issue with -lcuda path

hello there, here is output when installing

```g++ -I. -I./common -I./include -I./include/CL -I./otherarch -I./otherarch/tools -I./otherarch/sdcpp -I./otherarch/sdcpp/thirdparty -I./include/vulkan -O3 -DNDEBUG -std=c++11 -fPIC -DLOG_DISABLE_LOGS -D_GNU_SOURCE -pthread -s -Wno-multichar -Wno-write-strings -Wno-deprecated -Wno-deprecated-declarations -pthread -DGGML_USE_CUBLAS -DSD_USE_CUBLAS -I/usr/local/cuda/include -I/opt/cuda/include -I/targets/x86_64-linux/include ggml_v4_cublas.o ggml_v3_cublas.o ggml_v2_cublas.o ggml_v1.o expose.o common.o gpttype_adapter_cublas.o ggml-quants.o ggml-alloc.o ggml-backend.o llava.o llavaclip.o unicode.o grammar-parser.o sdcpp_cublas.o ggml-cuda.o ggml_v3-cuda.o ggml_v2-cuda.o ggml_v2-cuda-legacy.o -shared -o koboldcpp_cublas.so -lcuda -lcublas -lcudart -lcublasLt -lpthread -ldl -lrt -L/usr/local/cuda/lib64 -L/opt/cuda/lib64 -L/targets/x86_64-linux/lib -L/usr/local/cuda/targets/aarch64-linux/lib -L/usr/local/cuda/targets/sbsa-linux/lib -L/usr/lib/wsl/lib 
/nix/store/j2y057vz3i19yh4zjsan1s3q256q15rd-binutils-2.41/bin/ld: cannot find -lcuda: No such file or directory
collect2: error: ld returned 1 exit status
make: *** [Makefile:575: koboldcpp_cublas] Error 1
error: builder for '/nix/store/ilylx30p0i9yc7q52pd3yicikzbn3m21-koboldcpp-libs-1.61.2.drv' failed with exit code 2
error: 1 dependencies of derivation '/nix/store/il77gqs8iqf6rv61bcakazr1y03qjmc6-koboldcpp-1.61.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/f3794zyy605g7vwrf2fyqpfp7ynx566n-koboldcpp-1.61.2_fish-completions.drv' failed to build
error: 1 dependencies of derivation '/nix/store/4p5nyifj4r124d1lc6z5z0wn5aj33zij-man-paths.drv' failed to build
error: 1 dependencies of derivation '/nix/store/sbhjqs88si7csnpdckrj7karsnz0mqyi-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/wav4p1brs2wr47270j0drrl6ay4wik2r-nixos-system-nixos-24.05pre604424.d8fe5e6c92d0.drv' failed to build
```

checked:
https://nixos.wiki/wiki/CUDA
i had to do cuda-fhs.nix way to make the binary working


if you needed help from the maintainers, they offer their help
https://discord.gg/kYSbJAhsgF
https://discord.com/channels/849937185893384223/849937402050379787/1223647072280907828
(all assuming it's not a me issue)

note:
i already have 
```
  # cuda support
  nixpkgs.config.cudaSupport = true;
```
enabled

--- a/pkgs/koboldcpp/default.nix
+++ b/pkgs/koboldcpp/default.nix
@@ -45,9 +45,7 @@ let
 
     nativeBuildInputs = lib.optionals cudaSupport [
       cudaPackages.cuda_nvcc
-      # TODO: Replace with autoAddDriverRunpath
-      # once https://github.com/NixOS/nixpkgs/pull/275241 has been merged
-      cudaPackages.autoAddOpenGLRunpathHook
+      cudaPackages.autoAddDriverRunpath
     ];
 
     buildInputs =
@@ -79,6 +77,11 @@ let
       ++ lib.optionals openclSupport [ "LLAMA_CLBLAST=1" ]
       ++ lib.optionals blasSupport [ "LLAMA_OPENBLAS=1" ]
       ++ lib.optionals vulkanSupport [ "LLAMA_VULKAN=1" ];
+
+    preBuild = lib.optionalString cudaSupport ''
+      substituteInPlace Makefile \
+        --replace "-lcuda " "-lcuda -L${cudaPackages.cuda_cudart.lib}/lib/stubs "
+    '';
 
     installPhase = ''
       runHook preInstall

